src/functions.js

```diff
@@ -36,43 +42,85 @@ function cities( capitals, formatter ){
   // such as 'Paris is the capital of France'
 
   // apply formatter to each object in capitals array and return an array of resulting sentences
+  //var capI = capitals[i];//console.log('capital[i] is', capI);//var item = capI[key];
+  //console.log('item is', item, 'key is', key);
+  /* //second approach
+ var a=[]
+  for (var i = 0; i < capitals.length; i++) {
+      a.push(formatter());
+  }
+  return a;
+  function formatter (city, country) {
+   return `${capitals[i]['city']} is the capital of ${capitals[i]['country']}`;
+  } */
+/* //map approach mine weird but still works!
+  var a=[]
+  for (var i = 0; i < capitals.length; i++) {
+      a.push(formatter(capitals[i]));
+  }
+  return a;
+  var formatter= 
+    capitals.map ((city, country)=> 
+   `${capitals[i]['city']} is the capital of ${capitals[i]['country']}`);
+}  */
+//best Lor solution i was waiting to see it!
+var resultArray = capitals.map(arrayElement => `${arrayElement.city} is the capital of ${arrayElement.country}`);
+return (resultArray);
 }
```

Comment:
In this solution, you're not making use of the formatter() callback provided in the starter code. Happy to discuss how you make use of this on Saturday.